### DISCUSSION TITLE
3.2.7 ./prepare-source error (patches issue, solved)

### DISCUSSION THREAD

**Discussion by TonyTTTT** [RsyncProject/rsync#396] [Q&A] [answered]
./prepare-source gives error:
  autoconf -o configure.sh
  make: autoconf: No such file or directory
  make: *** [configure.sh] Error 1

macOS 13.0
**(No errors with 3.2.6)**

**Comment by WayneD** [comment]
If you download the release tar file, you don't need to have autoconf installed. But if you use the git repo, you need more helper tools. See https://github.com/WayneD/rsync/blob/master/INSTALL.md#how-to-build-and-install-rsync for what to install for a full from-git build.

**Reply by TonyTTTT** [reply]
I downloaded the 3.2.7 tar file from https://rsync.samba.org, and get the error when running ./prepare-source. 
I also tried skipping ./prepare-source, and starting with ./configure --disable-xxhash --disable-zstd --disable-lz4 
...and make gives the error:
autoconf -o configure.sh
make: autoconf: No such file or directory
make: *** [configure.sh] Error 1

I then downloaded the 3.2.6 tar file, and no errors

**Comment by TonyTTTT** [comment, marked as answer]
Update: I'm not sure why, but the error only occurs when I apply the patch: patch -p1 <patches/fileflags.diff
If I don't apply the fileflags patch, I can compile successfully 
Always applied the patch on prior releases without errors

**Reply by WayneD** [reply]
For some reason my master-branch dir ended up with ../m4/ paths in the aclocal.m4 file while the patch branches' build dirs got m4/ paths.  If you remove the patch chunk for the aclocal.m4 file from fileflags.diff, it should avoid the problem.

**Reply by TonyTTTT** [reply]
That fixes it.
